lua/hbac/setup.lua

```diff
@@ -1,18 +1,75 @@
 local command = require("hbac.command")
-local state   = require("hbac.state")
+local state = require("hbac.state")
 
-local M = {}
+local M = {
+	CONSTANTS = {
+		AUGROUP_AUTO_CLOSE = "hbac_autoclose",
+	},
+	opts = {
+		threshold = 10,
+		autoclose = true,
+	},
+}
 
 local id = vim.api.nvim_create_augroup("hbac", {
-	clear = false
+	clear = false,
 })
 
-M.setup = function(opts)
-	opts = opts or {
-		autoclose = true,
-		threshold = 10
-	}
+M.setup_autoclose = function()
+	state.autoclose_enabled = true
+	vim.api.nvim_create_autocmd({ "BufEnter" }, {
+		group = vim.api.nvim_create_augroup(M.CONSTANTS.AUGROUP_AUTO_CLOSE, { clear = true }),
+		pattern = { "*" },
+		callback = function()
+			local current_buf = vim.api.nvim_get_current_buf()
+			local buftype = vim.api.nvim_buf_get_option(current_buf, "buftype")
+			-- if the buffer is not a file - do nothing
+			if buftype ~= "" then
+				return
+			end
+
+			local buffers = vim.tbl_filter(function(buf)
+				local name = vim.api.nvim_buf_get_name(buf)
+				local listed = vim.api.nvim_buf_get_option(buf, "buflisted")
+				local windows_with_buf = vim.fn.win_findbuf(buf)
+				if name ~= "" and listed and #windows_with_buf == 0 then
+					return true
+				end
```

Comment:
This is really cool! So if a a buffer is shown in any window, it will not get closed, if I understand correctly. This was a "bug" that I wanted to fix myself, but you beat me to it!

Reply:
Exactly, along with this and `close_command` (new configuration opt) we have all the tools to avoid unwanted behavior